core/trino-main/src/main/java/io/trino/server/protocol/spooling/QueryDataEncoderSelector.java

```diff
@@ -16,15 +16,18 @@
 import java.util.Optional;
 
 /**
- * Responsible for choosing a QueryDataEncoder implementation based on a query Session.
+ * Responsible for choosing a {@link QueryDataEncoder.Factory} based on the encoding header
+ * which can contain multiple encodingIds separated by commas.
  * <p></p>
- * Important: needs to be stable across runs as it will be executed on multiple nodes participated in query execution
- * and output generation.
- * <p></p>
- * Returning Optional.empty() fallbacks to the direct protocol.
+ * Returning Optional.empty() fallbacks to the previous, direct protocol.
  */
 @FunctionalInterface
 public interface QueryDataEncoderSelector
 {
-    Optional<QueryDataEncoder.Factory> select(String encodingId);
+    Optional<QueryDataEncoder.Factory> select(String encodingHeader);
```

Comment:
Not a big fan of mixing header encoding with internal interface. I would rather pass List<String> as parameter

Reply:
Makes sense.